Swamp Sneak/Assets/Scripts/EnemyLineOfSight.cs

```diff
@@ -71,4 +71,27 @@ public class EnemyLineOfSight : MonoBehaviour {
 	public void SetSightDistance(int sd) {
 		sightDistance = sd;
 	}
+	
+	bool stealthCheck()
+	{
+
+		//Function to check if player is stealthed and also handle RNG factor
+		if(GameObject.Find("Player").GetComponent<ThirdPersonCharacter>().getStealth() == false)
+		{
+			return true;
+		}
+		
+		int chance = Random.Range(1,9);
+		print(chance);
+		
+		if (GameObject.Find("GameControl").GetComponent<GameControl>().playerStealth < chance)
+		{
+			return true;
```

Comment:
My comment above applies here as well. Also you could do a little trick here where you remove this whole if statement block and just have

`return (GameObject.Find("GameControl").GetComponent<GameControl>().playerStealth < chance)`